spec/factories.rb

```diff
@@ -67,6 +67,15 @@
     end
   end
 
+  factory :step_by_step_page_with_steps_and_links, parent: :step_by_step_page_with_steps do
```

Comment:
So I've played around with this a bit and adding the navigation rules to the `:step` creates a lot of errors. This is because I can't really just add a link but have to add the navigation rules. I think I'd be able to make it work by changing all of those tests but that would be a much bigger diff. Do you think it's worth doing for the sake of cleaner code?